File: src/infrastructure/entities/Product_Category.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn } from "typeorm";

import CategoryEntity from "./Category.entity";
import ProductEntity from "./Product.entity";

@Entity("product_categories")
export default class ProductCategoryEntity {
  @PrimaryGeneratedColumn("increment", { type: "bigint" })
  id: string;

  @ManyToOne(() => ProductEntity, { onDelete: "CASCADE" })
  @JoinColumn({ name: "product_id" })
  product: ProductEntity;

  @ManyToOne(() => CategoryEntity, { onDelete: "CASCADE" })
  @JoinColumn({ name: "category_id" })
  category: CategoryEntity;

  @Column({ type: "boolean", default: false, name: "is_primary" })
  isPrimary: boolean;
}

File: src/main/java/com/example/demo/product/entity/ProductCategoryEntity.java
package com.example.demo.product.entity;

import com.example.demo.category.entity.CategoryEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@Table(name = "product_categories")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class ProductCategoryEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.REMOVE)
    @JoinColumn(name = "product_id")
    private ProductEntity productEntity;
    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.REMOVE)
    @JoinColumn(name = "category_id")
    private CategoryEntity categoryEntity;
    @Column(name = "is_primary", columnDefinition = "BOOLEAN DEFAULT FALSE")
    private Boolean isPrimary;

    public static ProductCategoryEntity create(
            ProductEntity productEntity,
            CategoryEntity categoryEntity,
            Boolean isPrimary
    ) {
        return new ProductCategoryEntity(
                null,
                productEntity,
                categoryEntity,
                isPrimary
        );
    }
}
